perf-tool/src/server.cpp

```diff
@@ -0,0 +1,273 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "server.hpp"
+
+#include <arpa/inet.h>
+#include <iostream>
+#include <jvmti.h>
+#include <netinet/in.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <thread>
+#include <unistd.h>
+
+#include "agentOptions.hpp"
+#include <fstream>
+#include "perf.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using json = nlohmann::json;
+
+Server::Server(int portNo, const string commandFileName, const string logFileName)
+{
+    this->portNo = portNo;
+
+    if (commandFileName != "")
+    {
+        commandClient = new CommandClient(commandFileName);
```

Comment:
Since we allocate a new `CommandClient` object inside the constructor of `Server`, we need a implement a destructor that will `delete` `commandClient`